app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,53 @@
+class CartsController < ApplicationController
+  before_action :init_cart
+  before_action :select_item, except: :create
+  def create
+    @cart_params = params[:cart]
+    @cart.each do |line_item|
+      if line_item["book_id"] == @cart_params["book_id"].to_i
+        @current_item = line_item
+      end
+    end
+    if @current_item.present?
+      @current_item["quantity"] += @cart_params["quantity"].to_i
+    else
+      @cart << {
+        "book_id": @cart_params["book_id"].to_i,
+        "quantity": @cart_params["quantity"].to_i
+      }
```

Comment:
```suggestion
      @cart << {
        @cart_params["book_id"] =>@cart_params["quantity"].to_i
      }
```

chị nghĩ
cart lưu hash chỉ cần giá trị product_id làm key và quantity là value